src/Infrastructure/CachingPageRetriever.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+declare( strict_types = 1 );
+
+namespace WMDE\Fundraising\Frontend\Infrastructure;
+
+use Doctrine\Common\Cache\Cache;
+
+/**
+ * @licence GNU GPL v2+
+ * @author Jeroen De Dauw < jeroendedauw@gmail.com >
+ */
+class CachingPageRetriever implements PageRetriever {
+
+	private $pageRetriever;
+	private $cache;
+
+	public function __construct( PageRetriever $pageRetriever, Cache $cache ) {
+		$this->pageRetriever = $pageRetriever;
+		$this->cache = $cache;
+	}
+
+	public function fetchPage( string $pageName, string $fetchMode ): string {
+		if ( $this->cache->contains( $pageName ) ) {
+			return $this->cache->fetch( $pageName );
```

Comment:
The mode `render` is mostly used for retrieving displayable content. When using plain text mails or word lists, we don't want MediaWiki's parser to mess up the content. Word lists and also e. g. configuration files in JSON would break, the content of text mails would be "enriched" by HTML tags.